### ISSUE TITLE
Loss functions and fixes in backpropagation

### ISSUE THREAD

**Issue by JT4v4res** [JT4v4res/Neural-networks-and-deep-learning#20]
Just created the codes for the loss/cost functions, changed implementation of activation functions and fixed the bugs in the past multilayer network backpropgation.
